app/src/org/commcare/android/view/c3/AxisConfiguration.java

```diff
@@ -185,6 +181,11 @@ private JSONObject getAxis(String prefix) throws JSONException, InvalidStateExce
 
         JSONObject config = new JSONObject();
         boolean isX = prefix.equals("x");
+        
+        // X and primary Y axis show by default, but not secondary y. Force them all to show.
+        // Display secondary y axis, regardless of if it has data; this makes the
+        // whitespace around the graph look more reasonable.
+        config.put("show", true);

```

Comment:
the fix is that this code now happens after the `if (!showAxes)` check above, rather than happening outside of this function and ignoring `showAxes`.